### PR TITLE
API-8145: add github action for running unit tests

### PR DESCRIPTION
**Pull request by bkubiak**

### CODE REVIEW

File: .github/workflows/unittests.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Unit Tests
+on:
+  push:
+    branches: [ master ]
```
Comment (kogierman) [inline .github/workflows/unittests.yml:4]: I would add `v3.3` also here - or it needs to be a separate yml on that branch?

**Review by kogierman — APPROVED** [review]
👏

File: .github/workflows/unittests.yml (hunk 1 above)
Comment (bkubiak) [reply .github/workflows/unittests.yml:4]: I'm not sure, but I can cherry-pick the commit to v3.3, good idea

Comment (bkubiak) [reply .github/workflows/unittests.yml:4]: I wish we could use some regex here, like `v3.\d`

Comment (bkubiak) [reply .github/workflows/unittests.yml:4]: Oh, we can use a pattern (not exactly regex):

```
v3.[0-9]+
```

**Pull request merged**